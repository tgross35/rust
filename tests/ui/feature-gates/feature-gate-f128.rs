#![allow(unused)]

// ensure gating for primitive use
mod m1 {
    const A: f128 = 10.0; //~ ERROR the feature `f128` is unstable

    pub fn main() {
        let a: f128 = 100.0; //~ ERROR the feature `f128` is unstable
        let b = 0.0f128; //~ ERROR the feature `f128` is unstable
        foo(1.23);
    }

    fn foo(a: f128) {} //~ ERROR the feature `f128` is unstable

    struct Bar {
        a: f128, //~ ERROR the feature `f128` is unstable
    }
}

// ensure we don't restrict name as an identifier or custom types
mod m2 {
    #[allow(non_camel_case_types)]
    struct f128 {
        a: i32,
    }

    fn foo() {
        let f128 = ();
    }

    fn bar(a: f128) -> i32 {
        a.a
    }
}

fn main() {}
