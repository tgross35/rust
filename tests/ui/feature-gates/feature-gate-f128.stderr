error[E0658]: the feature `f128` is unstable
  --> $DIR/feature-gate-f128.rs:5:14
   |
LL |     const A: f128 = 10.0;
   |              ^^^^
   |
   = note: see issue #116909 <https://github.com/rust-lang/rust/issues/116909> for more information
   = help: add `#![feature(f128)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: the feature `f128` is unstable
  --> $DIR/feature-gate-f128.rs:8:16
   |
LL |         let a: f128 = 100.0;
   |                ^^^^
   |
   = note: see issue #116909 <https://github.com/rust-lang/rust/issues/116909> for more information
   = help: add `#![feature(f128)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: the feature `f128` is unstable
  --> $DIR/feature-gate-f128.rs:13:15
   |
LL |     fn foo(a: f128) {}
   |               ^^^^
   |
   = note: see issue #116909 <https://github.com/rust-lang/rust/issues/116909> for more information
   = help: add `#![feature(f128)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: the feature `f128` is unstable
  --> $DIR/feature-gate-f128.rs:16:12
   |
LL |         a: f128,
   |            ^^^^
   |
   = note: see issue #116909 <https://github.com/rust-lang/rust/issues/116909> for more information
   = help: add `#![feature(f128)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: the feature `f128` is unstable
  --> $DIR/feature-gate-f128.rs:9:17
   |
LL |         let b = 0.0f128;
   |                 ^^^^^^^
   |
   = note: see issue #116909 <https://github.com/rust-lang/rust/issues/116909> for more information
   = help: add `#![feature(f128)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0658`.
